pandas/core/groupby.py

```diff
@@ -1490,22 +1555,24 @@ def get_group_levels(self):
 
     _name_functions = {}
 
-    def _get_aggregate_function(self, how, values):
+    def _get_cython_function(self, kind, how, values, is_numeric):
 
         dtype_str = values.dtype.name
 
         def get_func(fname):
-            # find the function, or use the object function, or return a
-            # generic
+            # see if there is a fused-type version of function
+            # only valid for numeric
+            f = getattr(_algos, fname, None)
+            if f is not None and is_numeric:
+                return f
+
+            # otherwise find dtype-specific version, falling back to object
             for dt in [dtype_str, 'object']:
                 f = getattr(_algos, "%s_%s" % (fname, dtype_str), None)
                 if f is not None:
                     return f
 
-            if dtype_str == 'float64':
```

Comment:
as far as I can tell this was only here to handle `group_median` - which didn't have a dtype name in the function, but is float-only.  I just renamed the function to `group_median_float64`